db/migrate/20181210145601_create_subscriptions.rb

```diff
@@ -0,0 +1,14 @@
+class CreateSubscriptions < ActiveRecord::Migration[5.2]
+  def change
+    create_table :subscriptions, id: :uuid do |t|
+      t.string :email
+      t.integer :frequency
+      t.date :expires_on
+      t.integer :status, default: 0
+      t.jsonb :search_criteria
+      t.string :reference, null: false
```

Comment:
Latest amendment. Don't allow an empty reference at db level as we don't want to enforce model validation (as we will generate the value if the user does not set one).